Introduce chat subtitle

just-carlod opened the pull request:
Introduce chat subtite, this PR is still work in progress ...

marhkb commented on src/tdlib/chat.rs:
```diff
@@ -231,6 +243,38 @@ mod imp {
+                _ => unimplemented!(),
+            }
+        }
+
```
This should not be here, because we made all properties read-only some time ago. I think you re-committed it by accident.

just-carlod replied:
Right, I did a rebase and didn't notice it

just-carlod commented on src/strings.rs:
```diff
@@ -377,3 +377,350 @@ fn message_contact_registered(sender: &MessageSender) -> String {
     gettext_f("{sender} joined Telegram", &[("sender", &sender)])
 }
+
+pub(crate) fn user_status(status: UserStatus) -> String {
```
I've added your functions here (@melix99) I think it's better to place it here rather than in chat-window-dialog. Also lots of stuff introduced in this pr can be reused in #383

just-carlod commented on src/expressions.rs:
```diff
@@ -124,3 +125,82 @@ fn restriction_label_expression<T: glib::StaticType, V: glib::ToValue>(
     .upcast()
 }
+
+pub(crate) fn subtitle_expression(chat: &Chat) -> gtk::Expression {
```
this function is too much general and is likely going to be used only in one place, so maybe I should implement it directly in the chat-history

melix99 commented on src/strings.rs:
```diff
@@ -424,3 +424,314 @@ fn message_contact_registered(sender: &MessageSender) -> String {
     gettext_f("{sender} joined Telegram", &[("sender", &sender)])
 }
+
+pub(crate) fn chat_action(action: &ChatAction) -> String {
```
I want to port all of this away from the gettext macro before merging the PR. I'll comment here when I'll do it.

melix99 replied:
Done https://github.com/melix99/telegrand/commit/23f77ced6b17c921166fa3a7675381f8b2ce1a47.

krlade replied:
I thought the process of moving away from gettext macro is already completed

krlade replied:
Looks like gettext! is still used in some places

krlade committed:
sidebar-row: Tell tdlib to open/close chat

Tell tdlib to open selected and close unselected chat. This is needed
to recieve some updates e.g. ChatOnlineMemberCount

krlade committed:
chat: Add online-member-count property

krlade committed:
chat-history: Add chat subtitle

Added chat subtitle i.e. chat member count and online member count for
supergroups, subscribers count for channels and user status for private
chats
